beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -69,7 +69,15 @@ func (r *Service) processPendingBlocks(ctx context.Context) error {
 				"parentRoot":  hex.EncodeToString(b.Block.ParentRoot),
 			}).Info("Requesting parent block")
 			req := [][32]byte{bytesutil.ToBytes32(b.Block.ParentRoot)}
-			if err := r.sendRecentBeaconBlocksRequest(ctx, req, pids[rand.Int()%len(pids)]); err != nil {
+			pid := pids[rand.Int()%len(pids)]
+			for _, p := range pids {
+				if cs, _ := r.p2p.Peers().ChainState(p); cs != nil && cs.HeadSlot >= uint64(s) {
```

Comment:
A quick comment here would be super helpful for future readers -  I understand the context of this, but would be nice to have

Reply:
Done!